src/functions.php

```diff
@@ -72,7 +72,7 @@ function apply_type_map_to_document($document, array $typeMap)
  * @return string
  * @throws InvalidArgumentException
  */
-function generate_index_name($document)
+function generate_index_name($document): string
```

Comment:
I think you'll need one more pass with `phpcs`. I recall return type hints requiring a space on either side of the colon.

Reply:
That was changed to be in line with PSR-12, which prohibits this. (I've also had too many discussions about that rule back in the day, which is why I took the easy way of using PSR-12 instead)